backend/assignments/1/src/main/java/assignment_one/ExecuteTransaction.java

```diff
@@ -0,0 +1,193 @@
+package assignment_one;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Random;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class ExecuteTransaction implements Runnable {
+
+    private JsonNode jsonNode;
+    private static final String WALLET_ADDRESS = "wallet_address";
+    private static final Map<String, Lock> coinLocks = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String, Queue<JsonNode>> pendingBuyRequests = new ConcurrentHashMap<>();
+
+
+
+    public ExecuteTransaction(JsonNode jsonNode) {
+        this.jsonNode = jsonNode;
+    }
+
+    public ExecuteTransaction(){}
+
+    private String getBlockHash() {
+        String SALTCHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
+        StringBuilder transactionHash = new StringBuilder();
+        Random rnd = new Random();
+
+        for (double i = 0; i < 199999999; i++) {
+            i = i;
+        }
+        while (transactionHash.length() < 128) {
+            int index = (int) (rnd.nextFloat() * SALTCHARS.length());
+            transactionHash.append(SALTCHARS.charAt(index));
+        }
+        String hashCode = transactionHash.toString();
+        return "0x" + hashCode.toLowerCase();
+    }
+
+    private void processPendingBuyRequests(String coin) {
+        Queue<JsonNode> pendingRequests = pendingBuyRequests.get(coin);
+        Logging.logErr("Pending");
+        if (pendingRequests != null && !pendingRequests.isEmpty()) {
+            while (!pendingRequests.isEmpty()) {
+                JsonNode pendingRequest = pendingRequests.poll();
+                processBuy(pendingRequest, coin, pendingRequest.get("data").get(WALLET_ADDRESS).asText());
+            }
+        }
+    }
+
+
+    private void processBuy(JsonNode data, String coin, String walletAddress) {
```

Comment:
Add validations for all methods. And abstract them out.

Wrap them in a validator wrapper/function.
What if these checks are general and reusable?